rflx/pyrflx/typevalue.py

```diff
@@ -260,3 +460,350 @@ def accepted_type(self) -> type:
+        """
+
+        msg_as_bitstr: Bitstring = Bitstring().from_bytes(value)
+        self.assign_bitvalue(msg_as_bitstr, length)
+
+    def assign_bitvalue(self, value: Bitstring, length: int = 0) -> None:
+        """
+        :param value: Bitstring representation of the message
+        :param length: use length parameter if complete
+        length of the message is less than 1 Byte
+        """
+
+        current_field_name = self._next_field(model.INITIAL.name)
+        field_first_in_bitstr = 0
+        field_length = 0
+
+        while current_field_name != model.FINAL.name and (
+            field_first_in_bitstr + field_length
+        ) <= len(str(value)):
+
+            current_field = self._fields[current_field_name]
+            if isinstance(current_field.typeval, OpaqueValue) and not self._has_length(
+                current_field_name
+            ):
+
+                self._fields[current_field_name].first = self._get_first(current_field_name)
+                self.set(current_field_name, value[field_first_in_bitstr:])
+                self._fields[current_field_name].length = Number(current_field.typeval.length)
+            else:
+                assert isinstance(current_field.length, Number)
+                field_length = current_field.length.value
+
+                if field_length < 8:
+
+                    if length != 0 and length < 8:
+                        field_first_in_bitstr = field_first_in_bitstr + 8 - field_length
+
+                    self.set(
+                        current_field_name,
+                        value[field_first_in_bitstr : field_first_in_bitstr + field_length],
+                    )
+                    field_first_in_bitstr = field_first_in_bitstr + field_length
+
+                elif field_length >= 9 and field_length % 8 != 0:
+                    number_of_bytes = field_length // 8 + 1
+                    field_bits = Bitstring()
+
+                    for _ in range(number_of_bytes - 1):
+                        field_bits += value[field_first_in_bitstr : field_first_in_bitstr + 8]
+                        field_first_in_bitstr += 8
+
+                    assert isinstance(current_field.first, Number)
+                    k = field_length // number_of_bytes + 1
+                    field_bits += value[
+                        field_first_in_bitstr + 8 - k : current_field.first.value + field_length
+                    ]
+                    self.set(current_field_name, field_bits)
+
+                else:
+                    this_first = self._fields[current_field_name].first
+                    prev_first = self._fields[self._prev_field(current_field_name)].first
+                    assert isinstance(prev_first, Number)
+                    assert isinstance(this_first, Number)
+                    if prev_first.value == this_first.value:
+                        s = prev_first.value
+                    else:
+                        s = this_first.value
```

Comment:
It is needed in case current_field is a virtual node (e.g. Type_Length_TPID in ethernet). Otherwise the virtual node would be parsed and the next "real" node (which has the same first bit as the virtual node) would be parsed beginning with the last bit of the previous virtual node. The resulting message would be incorrect.

Reply:
In which case would the if-statement lead to another result than just using `s = this_first.value`?